From d0d7ea80c08338a084361bb625d62122456e69d9 Mon Sep 17 00:00:00 2001
From: Tao Huang <huangtao@rock-chips.com>
Date: Tue, 17 Mar 2020 19:21:21 +0800
Subject: [PATCH] net: netlink: Call netlink_proto_init() later when
 CONFIG_ROCKCHIP_THUNDER_BOOT=y

netlink_proto_init() depends on sock_init() to init sock_mnt.
But netlink_proto_init() and sock_init() are both in the same
core_initcall level.

Fix this by move netlink_proto_init() to the core_initcall_sync level.

Change-Id: I76733399c269efb191c5208fe45ce3add8993ebf
Signed-off-by: Tao Huang <huangtao@rock-chips.com>
---
 net/netlink/af_netlink.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/net/netlink/af_netlink.c b/net/netlink/af_netlink.c
index 1bb9f219f07d..b70de31b17c9 100644
--- a/net/netlink/af_netlink.c
+++ b/net/netlink/af_netlink.c
@@ -2779,4 +2779,8 @@ static int __init netlink_proto_init(void)
 	panic("netlink_init: Cannot allocate nl_table\n");
 }
 
+#ifdef CONFIG_ROCKCHIP_THUNDER_BOOT
+core_initcall_sync(netlink_proto_init);
+#else
 core_initcall(netlink_proto_init);
+#endif
-- 
2.35.3

